### ISSUE TITLE
Dataset 

### ISSUE THREAD

**Issue by paul0728** [HKUDS/KGRec#9]
作者您好,想再請教關於dataset的問題:
以last-fm為例,論文當中數據如圖
![image](https://github.com/HKUDS/KGRec/assets/44644609/966f27e8-d844-4735-8372-82ecc2c9bc7c)
1.interaction 數量對不上
論文中interactions數量為3034796
但是train.txt 之interaction 數量為1289003
test.txt 之interaction 數量為423635

2.entity
論文中entity數量為58266
但程式碼中n_entities數量為106389

3.dataset split 不一致
論文中寫說train/valid/test = 7:1:2
程式碼當中只有train/test = 1289003:423635 約為3:1

想知道為何會有如此不一致之情況?

4.test interval和early stop step設定的原因?
test_interval = 10 if args.dataset == 'last-fm' else 1
early_stop_step = 5 if args.dataset == 'last-fm' else 10

5.test time
我在last-fm執行kgrec 程式test time 跟train time 差不多久
在alibaba-fashion 執行kgrec程式test time是train time的3倍
請問該如何加速test time(參考您之前在mind執行的log,test time是遠小於train time的)

謝謝

**Comment by yuh-yang** [comment]
你好，这个数据集完全follow [KGIN](https://github.com/huangtinglin/Knowledge_Graph_based_Intent_Network)来的，之所以最后得到的interaction_mat的row_count是1289003，是因为原文件中有大量重复的interactions没有被过滤。n_entities在code中表示entity和item的总和。

test_interval和early_stop_step的设定完全是凭经验，保证baseline之间公平就好

因为test是CPU密集型任务，所以主要瓶颈还是在CPU，除了多线程可能没什么更好的加速方法

**Comment by yuh-yang** [comment]
你好，关于这一组数据集最早的描述应该来源于KGAT：

For each dataset, we randomly select 80% of interaction history of each user to constitute the training set, and treat the remaining as the test set. From the training set, we randomly select 10% of interactions as validation set to tune hyper-parameters.

这是一个7:1:2的setting，由于我们沿用了KGAT以及KGIN的setting，因此在相关部分也保持了同样的描述。实际上KGAT及后续的工作应该都没有val set，所以应该是类似于一个7:2或者简单7:3的setting。我们所有的数据和code都是开源的。

**Comment by paul0728** [comment]
感謝解說
